utils/atomics.go

```diff
@@ -0,0 +1,46 @@
+package utils
```

Comment:
Let's not have a utils package. See "Bad package names" in the [go blog](https://blog.golang.org/package-names).